contracts/satellite-chain/bridge-gateways/MaticXCAmpleRebaseGateway.sol

```diff
@@ -16,14 +16,12 @@ import {IXCAmple} from "../../_interfaces/IXCAmple.sol";
  *      the xc-ample contracts.
  *
  */
-contract MaticXCAmpleRebaseGateway is Layer2RebaseGateway, FxBaseChildTunnel {
```

Comment:
Initially assumed we could abstract all L2s to implement the same gateway interface, however there are operational differences between L2s which prevent this.

eg) Matic doesn't charge a ETH fee to send data from ethereum to matic but Arbitrum does. Arbitrum expects the caller to "pre-pay" the storage and execution cost of the cross chain transaction and thus has some additional parameters.

Leaning toward defining separate gateway interfaces for each bridge.

Reply:
Yeah. Unfortunately I think it's unlikely these bridges will be similar enough to abstract away completely.